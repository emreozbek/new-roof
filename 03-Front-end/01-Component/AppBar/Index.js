import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import { withStyles } from "material-ui/styles";
import {
  AppBar,
  Toolbar,
  IconButton,
  Typography,
  Menu,
  MenuItem
} from "material-ui";
import { Menu as MenuIcon, AccountCircle } from "material-ui-icons";

import styles from "./Style";

class AppBarComp extends Component {
  constructor() {
    super();
    this.onClick = this.HandleMenu.bind(this);
    this.state = {
      anchorEl: null,
      open: false
    };
  }
  HandleMenu() {
    this.props.HandleMenu(!this.props.menuState);
  }
  HandleClick = event => {
    this.setState({ anchorEl: event.currentTarget, open: true });
  };
  HandleClose = () => {
    this.setState({ anchorEl: null, open: false });
  };
  render() {
    const { classes, title, menuState, Logout } = this.props;
    const { anchorEl, open } = this.state;
    return (
      <AppBar
        className={classNames(classes.appBar, menuState && classes.appBarShift)}
      >
        <Toolbar disableGutters>
          <IconButton
            color="contrast"
            aria-label="open drawer"
            onClick={this.onClick}
          >
            <MenuIcon />
          </IconButton>
          <Typography
            type="title"
            color="inherit"
            className={classes.flex}
            noWrap
          >
            {title}
          </Typography>
          <div>
            <IconButton
              aria-owns={open ? "menu-appbar" : null}
              aria-haspopup="true"
              onClick={this.HandleClick}
              color="contrast"
            >
              <AccountCircle />
            </IconButton>
            <Menu
              id="menu-appbar"
              anchorEl={anchorEl}
              anchorOrigin={{
                vertical: "top",
                horizontal: "right"
              }}
              transformOrigin={{
                vertical: "top",
                horizontal: "right"
              }}
              open={open}
              onClose={this.HandleClose}
            >
              <MenuItem onClick={this.HandleClose}>Profile</MenuItem>
              <MenuItem onClick={Logout}>Logout</MenuItem>
            </Menu>
          </div>
        </Toolbar>
      </AppBar>
    );
  }
}
AppBarComp.propTypes = {
  title: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
  menuState: PropTypes.bool.isRequired,
  HandleMenu: PropTypes.func.isRequired,
  Logout: PropTypes.func.isRequired
};
export default withStyles(styles)(AppBarComp);
