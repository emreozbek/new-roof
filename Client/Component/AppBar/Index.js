import React, { Component } from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";
import classNames from "classnames";
import { AppBar, IconButton, Menu, MenuItem, Toolbar, Typography, withStyles } from "material-ui";
import { AccountCircle, Menu as MenuIcon } from "material-ui-icons";

import styles from "./Style";

class AppBarComp extends Component {
  constructor() {
    super();
    this.onClick = this.HandleMenu.bind(this);
    this.state = {
      anchorEl: null,
      open: false
    };
  }
  HandleMenu() {
    this.props.HandleMenu(!this.props.menuState);
  }
  HandleClick(event) {
    this.setState({ anchorEl: event.currentTarget, open: true });
  }
  HandleClose() {
    this.setState({ anchorEl: null, open: false });
  }
  render() {
    const { classes, title, menuState } = this.props;
    const { anchorEl, open } = this.state;
    return (
      <AppBar
        className={classNames(classes.appBar, menuState && classes.appBarShift)}
      >
        <Toolbar disableGutters>
          <IconButton
            color="contrast"
            aria-label="open drawer"
            onClick={this.onClick}
          >
            <MenuIcon />
          </IconButton>
          <Typography
            type="title"
            color="inherit"
            className={classes.flex}
            noWrap
          >
            {title}
          </Typography>
          <div>
            <IconButton
              aria-owns={open ? "menu-appbar" : null}
              aria-haspopup="true"
              onClick={this.HandleClick.bind(this)}
              color="contrast"
            >
              <AccountCircle />
            </IconButton>
            <Menu
              id="menu-appbar"
              anchorEl={anchorEl}
              anchorOrigin={{
                vertical: "top",
                horizontal: "right"
              }}
              transformOrigin={{
                vertical: "top",
                horizontal: "right"
              }}
              open={open}
              onClose={this.HandleClose.bind(this)}
            >
              <Link to="/profile">
                <MenuItem>Profile</MenuItem>
              </Link>
              <Link to="/logout">
                <MenuItem>Logout</MenuItem>
              </Link>
            </Menu>
          </div>
        </Toolbar>
      </AppBar>
    );
  }
}
AppBarComp.propTypes = {
  title: PropTypes.string,
  classes: PropTypes.object.isRequired,
  menuState: PropTypes.bool,
  HandleMenu: PropTypes.func
};
AppBarComp.defaultProps = {
  title: "",
  menuState: false,
  HandleMenu: () => {}
};

export default withStyles(styles)(AppBarComp);
