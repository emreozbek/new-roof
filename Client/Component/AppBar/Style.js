const styles = theme => ({
  acccountAlign: {
    display: "flex",
    justifyContent: "flex-end"
  },
  flex: {
    flex: 1
  },
  appBar: {
    transition: theme.transitions.create(["width", "margin"], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen
    })
  },
  appBarShift: {
    marginLeft: 240,
    width: `calc(100% - ${240}px)`,
    transition: theme.transitions.create(["width", "margin"], {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen
    })
  }
});

export default styles;
