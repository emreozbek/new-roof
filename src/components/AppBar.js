import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";
import { withStyles } from "material-ui/styles";
import { AppBar, Toolbar, IconButton, Typography } from "material-ui";
import MenuIcon from "material-ui-icons/Menu";
import AccountCircle from "material-ui-icons/AccountCircle";

import styles from "./styles/AppBar";

class AppBarComponent extends Component {
  constructor() {
    super();
    this.onClick = this.handleMenu.bind(this);
  }
  handleMenu() {
    this.props.handleMenu(!this.props.menuState);
  }
  render() {
    const { classes, title, menuState } = this.props;
    return (
      <AppBar
        className={classNames(classes.appBar, menuState && classes.appBarShift)}
      >
        <Toolbar disableGutters>
          <IconButton
            color="contrast"
            aria-label="open drawer"
            onClick={this.onClick}
          >
            <MenuIcon />
          </IconButton>
          <Typography
            type="title"
            color="inherit"
            className={classes.flex}
            noWrap
          >
            {title}
          </Typography>
          <IconButton aria-haspopup="true" color="contrast">
            <AccountCircle />
          </IconButton>
        </Toolbar>
      </AppBar>
    );
  }
}
AppBarComponent.propTypes = {
  title: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
  menuState: PropTypes.bool.isRequired,
  handleMenu: PropTypes.func.isRequired
};
export default withStyles(styles)(AppBarComponent);
