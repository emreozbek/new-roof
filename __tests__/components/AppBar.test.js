import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import AppBar from "../../Client/Component/AppBar/Index";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const wrapper = shallow(<AppBar />);
  expect(wrapper).toMatchSnapshot();
});
