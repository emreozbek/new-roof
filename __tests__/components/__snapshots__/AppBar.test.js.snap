// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<AppBarComp
  HandleMenu={[Function]}
  Logout={[Function]}
  classes={
    Object {
      "acccountAlign": "AppBarComp-acccountAlign-1",
      "appBar": "AppBarComp-appBar-3",
      "appBarShift": "AppBarComp-appBarShift-4",
      "flex": "AppBarComp-flex-2",
    }
  }
  menuState={false}
  title=""
/>
`;
